TheSadRogue.Primitives/Point.cs

```diff
@@ -525,6 +525,13 @@ public static (int x, int y) operator /(Point c, (int x, int y) tuple)
         [Pure]
         public bool Equals((int x, int y) other) => X == other.x && Y == other.y;
         #endregion
+
+        #region polar coordinates
+        public PolarCoordinate ToPolarCoordinate()
```

Comment:
Annotation that helps compilers/analyzers for structs, indicates the function does not change members:
```suggestion
        [Pure]
        public PolarCoordinate ToPolarCoordinate()
```